File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.py
def getGCD(n1, n2):
    return n1 if n2 == 0 else getGCD(n2, n1 % n2);

def solution(numer1, denom1, numer2, denom2):
    [numer, denom] = [(numer1 * denom2) + (numer2 * denom1),(denom1 * denom2)];
    gcd = getGCD(numer, denom);
    return [numer/gcd, denom/gcd]

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.js
function getGCD(n1, n2){
    if(n2 === 0) return n1;
    return getGCD(n2, n1 % n2)
}


function solution(numer1, denom1, numer2, denom2) {
    const [numer, denom] = [(numer1 * denom2) + (numer2 * denom1) , (denom1 * denom2)];
    const gcd = getGCD(numer, denom);
    return [numer/gcd, denom/gcd]
}